packages/editor/src/editor/browser/data-class/variable-tree-data.ts

```diff
@@ -54,6 +54,18 @@ export const variableTreeData = () => {
   return { of, loadChildrenFor, typesOfParam };
 };
 
+export const collectNodesWithChildren = (nodes: BrowserNode<Variable>[], parent?: string): BrowserNode<Variable>[] => {
+  let result: BrowserNode<Variable>[] = [];
+  for (const node of nodes) {
+    if (node.children && node.children.length > 0) {
+      const adjustedNode = { ...node, value: parent ? `${parent}.${node.value}` : node.value };
+      result.push(adjustedNode);
+      result = result.concat(collectNodesWithChildren(adjustedNode.children, adjustedNode.value));
```

Comment:
why not `result = [adjustedNode, ...collectNodesWithChildren(adjustedNode.children, adjustedNode.value)];`
IMO much more readable.